tests/docker-images/java-test-functions/pom.xml

```diff
@@ -20,90 +20,82 @@
 -->
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-    <parent>
-        <groupId>org.apache.pulsar.tests</groupId>
-        <artifactId>docker-images</artifactId>
-        <version>2.9.0-SNAPSHOT</version>
-    </parent>
-    <modelVersion>4.0.0</modelVersion>
-    <artifactId>java-test-functions</artifactId>
-    <name>Apache Pulsar :: Tests :: Docker Images :: Java Test Functions</name>
-    <dependencies>
-        <dependency>
-            <groupId>org.apache.pulsar</groupId>
-            <artifactId>pulsar-io-core</artifactId>
-            <version>${project.version}</version>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.pulsar</groupId>
-            <artifactId>pulsar-client</artifactId>
-            <version>${project.version}</version>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.avro</groupId>
-            <artifactId>avro</artifactId>
-            <version>${avro.version}</version>
-        </dependency>
-
-        <dependency>
-            <groupId>com.fasterxml.jackson.core</groupId>
-            <artifactId>jackson-databind</artifactId>
-            <version>${jackson.databind.version}</version>
-        </dependency>
-    </dependencies>
-    <packaging>jar</packaging>
+  <parent>
+    <groupId>org.apache.pulsar.tests</groupId>
+    <artifactId>docker-images</artifactId>
+    <version>2.9.0-SNAPSHOT</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+  <artifactId>java-test-functions</artifactId>
+  <name>Apache Pulsar :: Tests :: Docker Images :: Java Test Functions</name>
+  <dependencies>
+    <dependency>
+      <groupId>org.apache.pulsar</groupId>
+      <artifactId>pulsar-io-core</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.pulsar</groupId>
+      <artifactId>pulsar-functions-api</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <!-- allow sinks to use native Avro and Json -->
+    <dependency>
+      <groupId>org.apache.avro</groupId>
```

Comment:
this is not needed anymore, can you please drop this?

Reply:
I'm just wondering how the dependencies are made available if they are dropped. The current code in java-test-functions needs these libraries. 

It should be possible to build functions that depend just on pulsar-functions-api and connectors that depend on pulsar-io-core. There should be no dependency on pulsar-client library directly. pulsar-client-api dependency comes via pulsar-functions-api.

How would you resolve this conflict?

Reply:
these imports would fail if the dependencies would be removed: https://github.com/apache/pulsar/blob/767d2a4bf46c74f871ed35e776fdc4ff4119da72/tests/docker-images/java-test-functions/src/main/java/org/apache/pulsar/tests/integration/io/TestGenericObjectSink.java#L21-L23

Reply:
sorry, I missed "scope = provided" below